src/unix/thread.c

```diff
@@ -72,10 +75,59 @@ int uv_thread_create(uv_thread_t *tid, void (*entry)(void *arg), void *arg) {
 }
 
 
+int uv_thread_create_on(int proc_num,
+                        uv_thread_t *tid,
+                        void (*entry)(void *arg),
+                        void *arg) {
```

Comment:
> TBD if expressed in bits or bytes

The reason I bring that up is that using bytes could help future-proof the libuv API.  If a platforms grows a system API that goes beyond the binary "can or can't run on this CPU", bytes have more room for expansion.

For example, I imagine that on a NUMA system, you want to be able to tell the kernel that threads should ideally run on these N cores.  If that's not possible, then try these M cores and if all else fails, use P cores.